go/immudb-formula1/dataloader.go

```diff
@@ -0,0 +1,260 @@
+		cast: map[int]int{0:isInt, 1:isInt, 2:isInt, 3:isInt, 4:isInt, 5:isInt, 6:isInt,
+			8:isInt, 9:isInt, 10:isInt, 12:isInt, 13:isInt, 14:isInt, 16:isInt, 17:isInt},
+	},
+	"seasons": t_metadata{
+		create: "CREATE TABLE seasons(year INTEGER, url VARCHAR, PRIMARY KEY year)",
+		cast: map[int]int{0:isInt},
+	},
+	"status": t_metadata{
+		create: "CREATE TABLE status(statusId INTEGER, status VARCHAR, PRIMARY KEY statusId)",
+		cast: map[int]int{0:isInt},
+	},
+}
+
+
+func str_clean(s string) string {
+	s1 := strings.ToValidUTF8(s,string([]rune{unicode.ReplacementChar}))
+	s2 := strings.ReplaceAll(s1, "%", "%%")
+	s3 := strings.ReplaceAll(s2, "'", ".")
+	return s3
+}
+
+func valstring(name string, record []string) string {
+	var t []string
+	mdata:=metadata[name]
+	for i,field := range record {
+		castType, ok := mdata.cast[i]
+		if !ok {
+			castType=isString
+		}
+		switch castType {
+			case isString:
+				t = append(t, fmt.Sprintf("'%s'",str_clean(field)))
+			case isInt:
+			ii := 0.0
+			if field == "NULL" {
+				t=append(t, "NULL")
+				break
+			}
+			if field != "" {
+				var err error
+				ii, err = strconv.ParseFloat(field, 64)
+				if err != nil {
+					log.Printf("FIELDS: %v",record)
+					log.Printf("Unable to convert field %s [%d]: %s", field, i, err.Error())
+					ii = -1.0
+				}
+			}
+			t = append(t, strconv.Itoa(int(ii)))
+			case isTimestamp:
+				t = append(t, fmt.Sprintf("CAST('%s' AS TIMESTAMP)",field))
+		}
+	}
+	return strings.Join(t,",")
+}
+func load_table(client immudb.ImmuClient, ctx context.Context, name string) {
+	filename := fmt.Sprintf("CSV/%s.csv", name)
+	f, err := os.Open(filename)
+	if err != nil {
+		log.Fatalf("Unable to open file %s: %s", filename, err.Error())
+	}
+	r := csv.NewReader(f)
+
+	columns, err := r.Read()
+	if err != nil {
+		log.Fatalln(err)
+	}
+	column_string := strings.Join(columns, ",")
+
+	tx, err := client.NewTx(ctx)
+	if err != nil {
+		log.Fatalf("Load Table %s. Error while creating transaction: %s", name, err)
+	}
+	err = tx.SQLExec(ctx, metadata[name].create, nil)
+	if err != nil {
+		log.Fatalf("Load Table %s. Error while creating table: %s", name, err)
+	}
+
+	tx_count :=1 // the create table is a valid instruction
+
+        if name == "results" {
+	   tx.SQLExec(ctx, "CREATE INDEX ON results(driverId);CREATE INDEX ON results(statusId);", nil)
+           tx_count++
+        }
+
+        for {
+		record, err := r.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Fatalln(err)
+		}
+		value_string := valstring(name, record)
+		qstring := fmt.Sprintf("INSERT INTO %s(%s) VALUES (%s)", name, column_string, value_string)
```

Comment:
parameters may be used as well e.g. tx.SQLExec(ctx, "INSERT INTO @table(@col) VALUES (@val)", params)